sdk/core/azure-core-http-vertx/src/main/java/com/azure/core/http/vertx/implementation/BufferedVertxHttpResponse.java

```diff
@@ -13,38 +13,28 @@
 
 import java.nio.ByteBuffer;
 
-public final class BufferedVertxHttpResponse extends VertxHttpAsyncResponse {
+public final class BufferedVertxHttpResponse extends VertxHttpResponseBase {
 
-    private final Buffer body;
+    private final byte[] body;
 
     public BufferedVertxHttpResponse(HttpRequest azureHttpRequest, HttpClientResponse vertxHttpResponse, Buffer body) {
         super(azureHttpRequest, vertxHttpResponse);
-        this.body = body;
+        this.body = body.getBytes();
     }
 
     @Override
     public BinaryData getBodyAsBinaryData() {
-        return BinaryData.fromBytes(body.getBytes());
+        return BinaryData.fromBytes(body);
     }
 
     @Override
     public Flux<ByteBuffer> getBody() {
-        return Flux.defer(() -> {
-            if (this.body.length() == 0) {
-                return Flux.empty();
-            }
-            return Flux.just(ByteBuffer.wrap(this.body.getBytes()));
-        });
+        return (body.length == 0) ? Flux.empty() : Flux.just(ByteBuffer.wrap(body));
```

Comment:
why no more defer?

Reply:
Defer is only necessary is the `Flux` or `Mono` creation is costly, such as making a request, and the processing should be deferred until subscribed. In this case all data is already available, so deferring is just adding overhead with no benefit.